xbmc/filesystem/AudioBookFileDirectory.cpp

```diff
@@ -55,20 +57,108 @@ bool CAudioBookFileDirectory::GetDirectory(const CURL& url,
   std::string title;
   std::string author;
   std::string album;
+  std::string desc;
+  std::vector<std::string> artistsort;
+  std::vector<std::string> tagdata;
+  std::vector<std::string> separators{" feat. ", " ft. ", " Feat. ", " Ft. ",  ";", ":",
+                                      "|",       "#",     "/",       " with ", "&"};
+  const std::string musicsep =
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_musicItemSeparator;
+  if (musicsep.find_first_of(";/,&|#") == std::string::npos)
+    separators.push_back(musicsep); // add custom music separator from as.xml
 
   const int end_time_m4b_file =
       m_fctx->streams[0]->duration * av_q2d(m_fctx->streams[0]->time_base);
   const int end_time_mka_file = m_fctx->duration * av_q2d(av_get_time_base_q());
+  const bool isAudioBook = url.IsFileType("m4b");
+  // Some tags are relevant to the whole album - these are read first
+  CMusicInfoTag albumtag;
 
   AVDictionaryEntry* tag=nullptr;
   while ((tag = av_dict_get(m_fctx->metadata, "", tag, AV_DICT_IGNORE_SUFFIX)))
   {
-    if (StringUtils::CompareNoCase(tag->key, "title") == 0)
-      title = tag->value;
-    else if (StringUtils::CompareNoCase(tag->key, "album") == 0)
-      album = tag->value;
-    else if (StringUtils::CompareNoCase(tag->key, "artist") == 0)
-      author = tag->value;
+    if (isAudioBook)
+    {
+      if (StringUtils::CompareNoCase(tag->key, "title") == 0)
+        title = tag->value;
+      else if (StringUtils::CompareNoCase(tag->key, "album") == 0)
+        album = tag->value;
+      else if (StringUtils::CompareNoCase(tag->key, "artist") == 0)
```

Comment:
doesn't get the split/join treatment? separate separators/advanced setting for audiobooks needed?

Reply:
No.  Audiobooks have limited tags right now and don't need any advanced handling.  Else I would have added that.